src/test/java/com/microsoft/sqlserver/jdbc/TestResource.java

```diff
@@ -63,6 +63,11 @@ protected Object[][] getContents() {
             {"R_ConnectionURLNull", "The connection URL is null."},
             {"R_connectionIsNotClosed", "The connection is not closed."},
             {"R_invalidExceptionMessage", "Invalid exception message"},
+            {"R_invalidClientSecret", "AADSTS7000215: Invalid client secret provided"},
+            {"R_invalidCertFields",
+                    "Error reading certificate, please verify the location of the certificate.signed fields invalid"},
```

Comment:
Is "certificate.signed" a typo?

Reply:
Nope not a typo. That's the exact error message to check for.